redhat-developer/developers.redhat.com
# [DO NOT MERGE] [RHDENG-2753] Integrate Drupal with SSO login for content editors

This PR introduces SSO sign-in for content editors. In the first instance it is all direct members of  the RHDP dev team that will be asked to try this out.

This PR makes the following changes to Drupal:

- It incorporates the changes from #2796 to prevent SSO failures from caching of HTTP GET requests
- It installs and configures integration with KeyCloak for local dev and PR environments
- It installs and configures an active session limit for each user
   - Each user may now have a maximum of 3 active sessions

**Please note this PR is targeting `stage` branch so that we can test things there first.**

### JIRA Issue Link
* https://issues.jboss.org/browse/RHDENG-2753

### Verification Process

* Build should go green
* Verify that there are no pending configuration changes to import in the Drupal Admin interface
* Verify that the session limit module is installed and maximum permitted sessions is 3 per-user
* All of the following team members must confirm that they can login with their RHDP stage account:
  * @paulrobinson 
  * @robpblake 
  * @RubyGemTSL 
  * @LightGuard 
  * @jordanpagewhite 
  * @alistairmc 
  * @adelasofia 

#### For PR reviewers

Please visit the `/user/login` page for Drupal on this PR. You'll notice that there is a "Login with your developer account" section. Can you please test that you can log in using the account that you can login with for the front-end at https://developers.stage.redhat.com.

The account you are using for https://developers.stage.redhat.com must have an email address that matches your account in Drupal. For everyone this should be your `@redhat.com` address.

Once you're logged into Drupal, please just have a small play around e.g. edit some content, try and change some settings just to make sure things are working as expected.

Any questions, please let me know. Can you also comment or +1 this PR once you've confirmed that you can login.
